### PR TITLE
Move stub package to its own folder

### PR DESCRIPTION
**Pull request by mahesh-hegde**
Just a series of git renames.

### CODE REVIEW

**Comment by mahesh-hegde** [comment]
Update: I realized I forgot to modify the .github/workflows to run in jnigen/ directory instead; what is an idiomatic way to do it?

File: .github/workflows/test-package.yml [hunk 1]
```diff
@@ -64,10 +64,12 @@ jobs:
         run: dart pub get
       - name: Run VM tests
         run: dart test --platform vm
+      - name: Install coverage
+        run: dart pub global activate coverage
       - name: Collect coverage
-        run: ./tool/coverage.sh
```
Comment (dcharkes) [inline .github/workflows/test-package.yml:62]: Then we should also delete that fine, not move it.

Comment (mahesh-hegde) [reply .github/workflows/test-package.yml:62]: Haha newbie mistake. Fixed it.

**Comment by mahesh-hegde** [comment]
Is it sufficient to `grep -r` for jni_gen when renaming the package? I think it takes some attempts to get it right.

File: .github/workflows/test-package.yml [hunk 2]
```diff
@@ -43,6 +46,9 @@ jobs:
   test:
     needs: analyze
     runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: ./jnigen
```
Comment (dcharkes) [inline .github/workflows/test-package.yml:51]: here's another one for jni_gen

**Comment by mahesh-hegde** [comment]
Sorry too many little mistakes.

File: .github/workflows/test-package.yml [hunk 3]
```diff
@@ -21,6 +21,9 @@ jobs:
   # against Dart stable.
   analyze:
```
Comment (dcharkes) [inline .github/workflows/test-package.yml:22]: We will have two sub-packages jni_gen and jni.
We should analyze and test both.

So we should either (a) call this analyze_jni_gen or (b) move the `working-directory` to each run step, so we can have a different working directory for analyzing jni.

Though we can also address this with the PR adding `jni`.

**Comment by dcharkes** [comment]
> Sorry too many little mistakes.

Don't worry about it!

File: .github/workflows/test-package.yml (hunk 3 above)
Comment (mahesh-hegde) [reply .github/workflows/test-package.yml:22]: working-directory is per-job it seems. I think I will add an analyze_jni in the commit adding jni. jni_gen is the main part anyway.

Comment (mahesh-hegde) [reply .github/workflows/test-package.yml]: Nvm. Renamed it for uniformity.

**Comment by dcharkes** [comment]
> The workflow is not valid. .github/workflows/test-package.yml (Line: 47, Col: 12): Job 'test_jni_gen' depends on unknown job 'analyze'.

**Comment by mahesh-hegde** [comment]
Statically typed YAML when? :P

**Comment by dcharkes** [comment]
Well, you'll be a Dart package CI expert in no-time this way 😄

**Review by dcharkes — APPROVED** [review]
LGTM!

**Comment by devoncarew** [comment]
Hi - as part of this change, we'll also want to delete the CONTRIBUTING.md file, move the authors file to the top level of the repo, and copy the license file to the top level (in addition to creating a top-level readme file).

Here's an example of a top-level readme file for a monorepo: https://github.com/google/protobuf.dart. It has a brief description of the repo and uses a markdown table to direct people to the individual contained packages.

**Comment by dcharkes** [comment]
Oops, looks like the contributing.md got revived by the merge.

@mahesh-hegde would you please make a follow up PR to address these comments.

**Comment by mahesh-hegde** [comment]
While we are at it, should I also link the proposal from README?

**Pull request merged**